src/struc/model.jl

```diff
@@ -0,0 +1,826 @@
+with the lowest alternative location ID in alphabetical order is chosen."""
+function choosedefaultaltlocid(atom_one::Atom, atom_two::Atom)
+    if getoccupancy(atom_one) > getoccupancy(atom_two) ||
+            (getoccupancy(atom_one) == getoccupancy(atom_two) &&
+            Int(getaltlocid(atom_one)) < Int(getaltlocid(atom_two)))
+        return getaltlocid(atom_one)
+    else
+        return getaltlocid(atom_two)
+    end
+end
+
+
+# These can possibly be put as a Union
+"""Organise elements into a `Model`."""
+organisemodel(chains::ChainList; model_number::Int=1) = organise(chains; model_number=model_number)
+organisemodel(residues::ResidueList; model_number::Int=1) = organise(organise(residues); model_number=model_number)
+organisemodel(atoms::AtomList; model_number::Int=1) = organise(organise(organise(atoms)); model_number=model_number)
+organisemodel(chain::Chain; model_number::Int=1) = organisemodel([chain]; model_number=model_number)
+organisemodel(res::AbstractResidue; model_number::Int=1) = organisemodel(AbstractResidue[res]; model_number=model_number)
+organisemodel(atom::AbstractAtom; model_number::Int=1) = organisemodel(AbstractAtom[atom]; model_number=model_number)
+
+
+"""Organise elements into a `Structure`."""
+organisestruc(models::ModelList; struc_name::AbstractString="") = organise(models; struc_name=struc_name)
+organisestruc(chains::ChainList; struc_name::AbstractString="", model_number::Int=1) = organise(organisemodel(chains; model_number=model_number); struc_name=struc_name)
+organisestruc(residues::ResidueList; struc_name::AbstractString="", model_number::Int=1) = organise(organisemodel(residues; model_number=model_number); struc_name=struc_name)
+organisestruc(atoms::AtomList; struc_name::AbstractString="", model_number::Int=1) = organise(organisemodel(atoms; model_number=model_number); struc_name=struc_name)
+organisestruc(model::Model; struc_name::AbstractString="") = organisestruc([model]; struc_name=struc_name)
+organisestruc(chain::Chain; struc_name::AbstractString="", model_number::Int=1) = organisestruc([chain]; struc_name=struc_name, model_number=model_number)
+organisestruc(res::AbstractResidue; struc_name::AbstractString="", model_number::Int=1) = organisestruc(AbstractResidue[res]; struc_name=struc_name, model_number=model_number)
+organisestruc(atom::AbstractAtom; struc_name::AbstractString="", model_number::Int=1) = organisestruc(AbstractAtom[atom]; struc_name=struc_name, model_number=model_number)
+
+
+"""Determines if an `AbstractAtom` is a non-hetero atom, i.e. came from an ATOM
+record."""
+stdatomselector(atom::AbstractAtom) = !ishetatom(atom)
+
+"""Determines if an `AbstractAtom` is a hetero atom, i.e. came from a HETATM
+record."""
+hetatomselector(atom::AbstractAtom) = ishetatom(atom)
+
+"""Determines if an `AbstractAtom` has its atom name in the given `Array`."""
+atomnameselector(atom::AbstractAtom, atom_names::Array{AbstractString,1}) = getatomname(atom) in atom_names
```

Comment:
This membership checking is linear time. You can use the `Set` type to make it constant time.